src/librustc_typeck/astconv.rs

```diff
@@ -1540,6 +1560,38 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx>+'o {
         result_ty
     }
 
+    pub fn ast_const_to_const(
+        &self,
+        ast_const: &hir::ConstArg,
+        ty: Ty<'tcx>
+    ) -> &'tcx ty::Const<'tcx> {
+        debug!("ast_const_to_const(id={:?}, ast_ty={:?})", ast_const.value.id, ast_const);
+
+        let tcx = self.tcx();
+        let def_id = tcx.hir.local_def_id(ast_const.value.id);
+
+        let mut val = ConstValue::Unevaluated(def_id, Substs::identity_for_item(tcx, def_id));
+
+        match tcx.describe_def(def_id) {
+            Some(Def::ConstParam(def_id)) => {
```

Comment:
How would this work? The `def_id` is that of the `AnonConst`, it can't be a parameter.
Maybe you want instead to check whether the expression *inside* `AnonConst` is an `ExprKind::Path` that has the `Def::ConstParam` resolution?